app/services/legacy/dataset_import_service.rb

```diff
@@ -84,21 +85,36 @@ def create_resource_base_attributes(resource, dataset)
   end
 
   def create_datafile_date_attributes(resource)
-    if resource["date"].blank?
-      {}
-    else
-      dates = get_start_end_date(resource["date"])
-      start_date = Date.parse(dates[0])
-      end_date = Date.parse(dates[1])
-
-      {
-        start_date: start_date,
-        end_date: end_date,
-        day: end_date.day,
-        month: end_date.month,
-        year: end_date.year
-      }
-    end
+    return {} if resource['date'].blank?
+
+    dates = get_start_end_date(resource['date'])
+    start_date = parse_date(dates[0])
+    end_date = parse_date(dates[1])
+
+    return {} if start_date.nil? || end_date.nil?
+    
+    date_attributes = {
+      start_date: start_date,
+      end_date: end_date
+    }
+
+    day_month_year = day_month_year_from(end_date)
+    date_attributes.merge(day_month_year)
+  end
+
+  def parse_date(date)
+    Date.parse(date)
+  rescue ArgumentError
```

Comment:
Check with @laurentqro, but I thought the correct syntax for exception handling in Ruby would be to wrap the code that might raise the exception in a `begin..rescue..end` block